src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py

```diff
@@ -551,6 +551,7 @@ def get_timesteps(self, num_inference_steps, strength, device):
 
         t_start = max(num_inference_steps - init_timestep, 0)
         timesteps = self.scheduler.timesteps[t_start * self.scheduler.order :]
+        self.scheduler._step_index_init = t_start * self.scheduler.order
```

Comment:
This needs to be a very general function call (e.g. one that every scheduler has).

We could maybe call it `self.scheduler.set_begin_index(...)` and then make sure that every scheduler has such a function.